src/telegram/process_update_task.rs

```diff
@@ -25,15 +26,90 @@ impl ProcessUpdateTask {
         Self { update }
     }
 
-    pub async fn process(&self) {
-        // TODO: process users messages here
-        // Create a ApiClient
+    pub async fn process(&self) -> Result<(), BotError> {
+        log::info!("Received a message {:?}", self.update);
+
         let api = AsyncApi::new(&RUST_TELEGRAM_BOT_TOKEN);
-        let opwm_token = &OPEN_WEATHER_MAP_API_TOKEN;
+        // for now with _ to pass warnings
+        let _opwm_token = &OPEN_WEATHER_MAP_API_TOKEN;
         let keypair = Rsa::private_key_from_pem(&BINARY_FILE).unwrap();
         let keypair = PKey::from_rsa(keypair).unwrap();
 
-        log::info!("Received a message {:?}", self.update);
+        if let UpdateContent::Message(message) = &self.update.content {
+            let (chat_id, user_id, user) = Self::get_info_from_message(message);
+
+            Self::send_typing(message, &api).await?;
```

Comment:
I think typing can be outside of if. and you can return something for every message

if you can not process a message (the state is wrong), you can just print an error message for a user

Reply:
Not relevant for now